.NET/Microsoft.Recognizers.Text.DateTime/Extractors/BaseDateExtractor.cs

```diff
@@ -306,9 +306,16 @@ private List<Token> DurationWithBeforeAndAfter(string text, DateObject reference
             foreach (var er in durationEr)
             {
                 // if it is a multiple duration and its type is not equal to Date than skip it.
-                if (er.Data != null && er.Data.ToString() != Constants.MultipleDuration_Date)
+                if (er.Data != null && er.Data.ToString().StartsWith("multipleDuration") && er.Data.ToString() != Constants.MultipleDuration_Date)
                 {
                     continue;
+                }
+
+                // special treatment for cases like "more than 3 days from today"

```

Comment:
For cases like "more than 3 days from today", this method should extract "3 days from today" (duration).
But the DurationExtractor will extract "more than 3 days" first.
Here we add special treatment for this "more than 3 days" to let this method work as expected.
This is not quite beautiful here, but seems no good way.